frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/request-vehicles/simulated-region-overview-behavior-request-vehicles.component.ts

```diff
@@ -0,0 +1,161 @@
+    styleUrls: [
+        './simulated-region-overview-behavior-request-vehicles.component.scss',
+    ],
+})
+export class RequestVehiclesComponent implements OnChanges {
+    @Input() simulatedRegionId!: UUID;
+    @Input() requestBehaviorId!: UUID;
+
+    requestBehaviorState$!: Observable<RequestBehaviorState>;
+
+    requestTargetOptions$!: Observable<{
+        [key in RequestTargetOption]: string;
+    }>;
+
+    waitingForAnswer$!: Observable<boolean>;
+
+    nextTimeoutIn$!: Observable<number>;
+
+    initialRequestTargetOption$!: Observable<RequestTargetOption>;
+
+    constructor(
+        private readonly store: Store<AppState>,
+        private readonly exerciseService: ExerciseService
+    ) {}
+
+    ngOnChanges(): void {
+        this.requestTargetOptions$ = this.store
+            .select(selectSimulatedRegions)
+            .pipe(
+                map((simulatedRegions) => {
+                    const options = Object.entries(simulatedRegions)
+                        .map(([id, simulatedRegion]) => [
+                            id,
+                            simulatedRegion.name,
+                        ])
+                        .filter(([id, _name]) => id !== this.simulatedRegionId)
+                        .sort(([id1, name1], [id2, name2]) =>
+                            name1 === name2 ? 0 : name1! < name2! ? -1 : 1
+                        );
+                    options.unshift(['trainees', 'Die Trainierenden']);
+                    return Object.fromEntries(options);
+                })
+            );
+
+        this.requestBehaviorState$ = this.store.select(
+            createSelectBehaviorState(
+                this.simulatedRegionId,
+                this.requestBehaviorId
+            )
+        );
+
+        this.initialRequestTargetOption$ = this.requestBehaviorState$.pipe(
+            map((requestBehaviorState) => {
+                if (
+                    requestBehaviorState.requestTarget.type ===
+                    'traineesRequestTarget'
+                ) {
+                    return 'trainees';
+                }
+                return requestBehaviorState.requestTarget
+                    .targetSimulatedRegionId;
+            })
+        );
+
+        this.waitingForAnswer$ = this.requestBehaviorState$.pipe(
+            map((requestBehaviorState) =>
+                isWaitingForAnswer(requestBehaviorState)
+            )
+        );
+
+        const activities$ = this.store.select(
+            createSelectActivityStates(this.simulatedRegionId)
+        );
+
+        const currentTime$ = this.store.select(selectCurrentTime);
+
+        this.nextTimeoutIn$ = combineLatest([
+            this.requestBehaviorState$,
+            activities$,
+            currentTime$,
+        ]).pipe(
+            map(([requestBehaviorState, activities, currentTime]) => {
+                if (!requestBehaviorState.delayEventActivityId) return -1;
```

Comment:
You could use `isWaitingForTimeout` here.

Reply:
I've completely removed the deprecation in 103c3d3ddcd06afd35c9ce1aa574a9bf3fa3771e